### PR TITLE
docs: added References section

### PR DESCRIPTION
**Pull request by marcellorinaldo**
Signed-off-by: Marcello Martina <martina.marcello.rinaldo@outlook.com>

> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

This PR ports the References section from the old documentation.

**Related Issue:** N/A.

**Description of the solution adopted:** N/A.

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
@mattdibi can you please review?

File: docs/references/mqtt-namespace.md [hunk 1]
```diff
@@ -0,0 +1,1190 @@
...
+!!! note
+    For the scope of this document, some aspects concerning the encoding and compressing of the payload are not included.
+
+The applicability of the remote management solution, as inspired by the OSGi component model, can be extended beyond OSGi as the contract with the managing server based on MQTT topics and XML payloads.
+
+### Remote OSGi ConfigurationAdmin Interactions via MQTT
+
+An application bundle is installed in the gateway to allow for remote management of the configuration properties of the services running in the OSGi container.
+
+For information about the OSGi Configuration Admin Service and the OSGi Meta Type Service, please refer to the [OSGi Service Platform Service Compendium 4.3 Specifications](http://www.osgi.org/Specifications/HomePage).
```
Comment (mattdibi) [inline docs/references/mqtt-namespace.md]: 404 link

File: docs/references/mqtt-namespace.md [hunk 2]
```diff
@@ -0,0 +1,1190 @@
...
+    <ns2:property type="Integer" array="false" name="watchdog.timeout">
+      <ns2:value>10000</ns2:value>
+    </ns2:property>
+  </ns2:properties>
+</ns2:configuration>
+```
+
+The service configuration XML message is comprised of the following parts:
+
+* The **Object Class Definition** (OCD), which describes the service attributes that may be configured. (The syntax of the OCD element is described in the [OSGi Service Platform Service Compendium 4.3 Specifications](http://www.osgi.org/Specifications/HomePage), Section 105.3.)
```
Comment (mattdibi) [inline docs/references/mqtt-namespace.md]: 404 link

File: docs/references/mqtt-namespace.md [hunk 3]
```diff
@@ -0,0 +1,1190 @@
...
+  
+* Request Payload:
+    * Nothing application-specific beyond the request ID and requester client ID
+  
+* Response Payload:
+    * Configurations of all the registered services serialized in XML format
+
+#### Read Configuration for a Given Service
+
+This operation provides configurations for a specific service that is identified by an OSGi service persistent identifier (**pid)**.
```
Comment (mattdibi) [inline docs/references/mqtt-namespace.md]:
```suggestion
This operation provides configurations for a specific service that is identified by an OSGi service persistent identifier (**pid**).
```

File: docs/references/mqtt-namespace.md [hunk 4]
```diff
@@ -0,0 +1,1190 @@
...
+
+When one such service is selected, a form is dynamically generated based on the metadata provided in the service OCD. This form includes logic to handle different attribute types, validate acceptable value ranges, and render optional values as drop-downs. When the form is submitted, the new values are communicated to the device through an MQTT resource update message.
+
+
+
+## Remote OSGi DeploymentAdmin Interactions via MQTT
+
+An application is installed in the gateway to allow for the remote management of the deployment packages installed in the OSGi container.
+
+For information about the OSGi Deployment Admin Service, please refer to the [OSGi Service Platform Service Compendium 4.3 Specifications](http://www.osgi.org/Specifications/HomePage).
```
Comment (mattdibi) [inline docs/references/mqtt-namespace.md]: 404 link

**Review by mattdibi — CHANGES_REQUESTED** [review]

**Review by mattdibi — APPROVED** [review]

**Pull request merged**